src/overlays/actors/ovl_Obj_Grass/z_obj_grass.c

```diff
@@ -28,41 +34,497 @@ const ActorInit Obj_Grass_InitVars = {
+    Vec3f spB0;
+    s32 i;
 
-extern ColliderCylinderInit D_809AAB20;
+    for (i = 0; i < ARRAY_COUNT(D_809AAB4C); i++) {
+        spB0.x = ptr->unk_00.x + (D_809AAB4C[i].x * 8.0f);
+        spB0.y = (ptr->unk_00.y + (D_809AAB4C[i].y * 8.0f)) + 10.0f;
+        spB0.z = ptr->unk_00.z + (D_809AAB4C[i].z * 8.0f);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Grass/func_809A9110.s")
+        spBC.x = (Rand_ZeroOne() - 0.5f) * 8.0f;
+        spBC.y = Rand_ZeroOne() * 10.0f;
+        spBC.z = (Rand_ZeroOne() - 0.5f) * 8.0f;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Grass/func_809A91FC.s")
+        EffectSsKakera_Spawn(globalCtx, &spB0, &spBC, &spB0, -100, 64, 40, 3, 0,
+                             D_809AAB7C[(s32)(Rand_ZeroOne() * 111.1f) & 7], 0, 0, 80, -1, 1, D_040527F0);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Grass/func_809A92D0.s")
+        spB0.x = ptr->unk_00.x + (D_809AAB4C[i].x * 16.0f);
+        spB0.y = (ptr->unk_00.y + (D_809AAB4C[i].y * 16.0f)) + 10.0f;
+        spB0.z = ptr->unk_00.z + (D_809AAB4C[i].z * 16.0f);
 
+        spBC.x = (Rand_ZeroOne() - 0.5f) * 6.0f;
+        spBC.y = Rand_ZeroOne() * 10.0f;
+        spBC.z = (Rand_ZeroOne() - 0.5f) * 6.0f;
+
+        EffectSsKakera_Spawn(globalCtx, &spB0, &spBC, &spB0, -100, 64, 40, 3, 0,
+                             D_809AAB7C[(s32)(Rand_ZeroOne() * 111.1f) % 7], 0, 0, 80, -1, 1, D_040528B0);
+    }
+}
+#else
+static Vec3f D_809AAB4C[] = {
+    { 0.0f, 0.707099974155f, 0.707099974155f },
+    { 0.707099974155f, 0.707099974155f, 0.0f },
+    { 0.0f, 0.707099974155f, -0.707099974155f },
+    { -0.707099974155f, 0.707099974155f, 0.0f },
+};
+static s16 D_809AAB7C[] = { 0x6C, 0x66, 0x60, 0x54, 0x42, 0x37, 0x2A, 0x26 };
+void func_809A9314(ObjGrassStruct1_1* ptr, GlobalContext* globalCtx);
 #pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Grass/func_809A9314.s")
+#endif
+
+void ObjGrass_Init(Actor* thisx, GlobalContext* globalCtx) {
+    ObjGrass* this = THIS;
+    s32 i;
+
+    Actor_SetScale(&this->actor, 0.4f);
+
+    for (i = 0; i < ARRAY_COUNT(this->unk_2948); i++) {
+        Collider_InitCylinder(globalCtx, &this->unk_2948[i].collider);
+        Collider_SetCylinder(globalCtx, &this->unk_2948[i].collider, &this->actor, &sCylinderInit);
+    }
+
+    this->actor.colChkInfo.mass = MASS_IMMOVABLE;
+    this->unk_3288 = (u32)Rand_Next() >> 0x10;
+    this->unk_328A = (u32)Rand_Next() >> 0x10;
+    this->unk_328C = (u32)Rand_Next() >> 0x10;
+    this->unk_328E = (u32)Rand_Next() >> 0x10;
+    this->unk_3290 = (u32)Rand_Next() >> 0x10;
+}
+
+void ObjGrass_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    ObjGrass* this = THIS;
+    s32 i;
+
+    for (i = 0; i < ARRAY_COUNT(this->unk_2948); i++) {
+        Collider_DestroyCylinder(globalCtx, &this->unk_2948[i].collider);
+    }
+
+    for (i = 0; i < ARRAY_COUNT(this->unk_3298); i++) {
+        ObjGrassUnkStruct** ptr = &this->unk_3298[i];
+
+        if ((*ptr) != NULL) {
```

Comment:
```suggestion
        if (*ptr != NULL) {
```
Don't think parens are necessary.